scripts/coverage.sh

```diff
@@ -42,7 +45,19 @@ RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov "${pac
 
 echo "--- grcov"
 
-_ grcov target/cov/debug/deps/ > target/cov/lcov-full.info
+# Create a clean room dir only with updated gcda/gcno files for this run
+data_dir=target/cov/tmp
+rm -rf "$data_dir"
+mkdir -p "$data_dir"
+
+find target/cov -name \*.gcda -newer "$timing_file" |
+  (while read -r gcda_file; do
+    gcno_file="${gcda_file%.gcda}.gcno"
+    ln -s "../../../$gcda_file" "$data_dir/$(basename "$gcda_file")"
```

Comment:
I guess `realpath` isn't available on macOS...

Reply:
yeah realpath is a GNU program that requires an extra install for macOS.   We can require it if absolutely needed but that's not ideal (another barrier for entry)

Reply:
I suggest just removing the `data_dir` variable and directly embedding `target/cov/tmp` everywhere `data_dir` is used.   The two `ln` command aren't generic enough to handle any modifications to `data_dir` due to the `../../..` so the variable doesn't really had much (and in fact could hurt a future person who attempts to modify this file).

Reply:
I think you have a nice point!: https://github.com/solana-labs/solana/pull/7576/commits/690a7b3f9c1f81a2c55a3740fee5bcfdb8bab600